### ISSUE TITLE
linted files using node_eslintrc.js don't need to declare node env at the top of the file

### ISSUE THREAD

**Issue by zepumph** [phetsims/chipper#962]
Discovered as part of https://github.com/phetsims/chipper/issues/811

We have over 100 usages of `/* eslint-env node */`, but if that files is covered under the set of lint rules defined by `node_eslintrc.js`, then this is superfluous. My guess is that most can be removed (like in perennial).

@ariel-phet, this is grunt work that would easy for a junior dev to do if they wanted to familiarize themselves with node at all

Basically they would search through all usages of `/* eslint-env node */` in the project, and if the file is nested in a directory that contains a `.eslintrc.js` file or a repo's package.json specifies that it extends from `node_eslintrc.js` (for example https://github.com/phetsims/chipper/blob/master/eslint/rules/.eslintrc.js), then this eslint-env line can be removed.

I'll keep myself assigned to address at low prioirty.

**Comment by muedli** [comment]
@zepumph, I believe I found and removed all remaining usages of `/* eslint-env node */` where the directory in which the usage resided contained a `.eslintrc.js` file or the repo in which the usage resided had a `package.json` such that the `eslintConfig` extended from `node_eslintrc.js`.

**Comment by muedli** [comment]
I should note that almost all of the usages I found while searching the whole project in IntelliJ were in Gruntfiles, and they didn't either of the criteria you listed for removal.

**Comment by zepumph** [comment]
This is great, thanks @muedli 

Searching for `-env` and excluding `Grunfile.js` files, I see 21 more usages. They mostly feel right to me.

@jonathanolson can you please speak to the reason why some MR files in perennial are declared as supporting both browser and node environments via `/* eslint-env browser, node */`. I see this for `Maintenance.js`, `ReleaseBranch.js`, `Patch.js`, and `ModifiedBranch.js`?

Aren't these just node files? My recommendation with my current level of understanding is to remove all of these declarations, and have them covered under the node_eslintrc.js file rules stated in PERENNIAL/package.json

**Comment by jonathanolson** [comment]
> Aren't these just node files?

Yes, it may have arisen from certain ES6 features not being allowed in the node.js rules? I'm fine with those declarations being removed, those should for now be node.js files only.

**Comment by zepumph** [comment]
Sounds good thanks!
